neurobooth_os/config.py

```diff
@@ -2,24 +2,31 @@
 """
     Ensures that the base neurobooth-os config file exists and makes config file available as neurobooth_config
 """
+import os.path
 import os.path as op
-from os.path import expanduser
 from os import environ
 import json
 
-# from neurobooth_os.logging import make_default_logger
+from neurobooth_os.util.constants import NODE_NAMES
 
-# if files does not exist raise an exception
-# logger = make_default_logger()
-
-# fname = op.join(expanduser("~"), ".neurobooth_os_config")
 fname = op.join(environ.get("NB_CONFIG"), "neurobooth_os_config.json")
-print(fname)
+
 if not op.exists(fname):
     msg = "Required config file does not exist"
-    # logger.critical(msg)
     raise IOError(msg)
 
-# TODO: Add another exception here?
 with open(fname, "r") as f:
     neurobooth_config = json.load(f)
+
+    destination = neurobooth_config["remote_data_dir"]
+    if not os.path.exists(destination):
+        raise FileNotFoundError(f"The remote_data_dir ({destination}) does not exist.")
+    if not os.path.isdir(destination):
+        raise IOError(f"The remote_data_dir ({destination}) is not a folder.")
+
+    for name in NODE_NAMES:
+        source = neurobooth_config[name]["local_data_dir"]
+        if not os.path.exists(source):
+            raise FileNotFoundError(f"The local_data_dir ({source}) for server {name} does not exist.")
+        if not os.path.isdir(source):
+            raise IOError(f"The local_data_dir ({source}) for server {name} is not a folder.")
```

Comment:
Would personally make a small `validate_folder` method (may be useful as we expand configs) and then do:
```
validate_folder(neurobooth_config["remote_data_dir"])
for name in NODE_NAMES:
    validate_folder(neurobooth_config[name]["local_data_dir"])
```
`source` and `destination` don't make sense in this more generic context of the config loader (which is agnostic to the data transfer).

Reply:
I implemented a compromise solution.

Reply:
Works for me!